### PR TITLE
Coverage++

### PR DESCRIPTION
**Pull request by Kriechi**
This PR completes the restructuring and linting checks from earlier PRs.
Mostly moving stuff around.

### CODE REVIEW

File: mitmproxy/contentviews/base.py [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
+from mitmproxy.net import http
+from mitmproxy.utils import strutils
+
 
+VIEW_CUTOFF = 512
 KEY_MAX = 30
 
+views = []  # type: List[View]
+content_types_map = {}  # type: Dict[str, List[View]]
+view_prompts = []  # type: List[Tuple[str, str]]
```
Comment (mhils) [inline mitmproxy/contentviews/base.py]: Not sure if this belongs together. `base` previously was stuff that was used by views only, whereas `__init__` contained the external API. Maybe move it to `viewmanager` or so?

File: mitmproxy/utils/bits.py [hunk 2]
```diff
@@ -1,13 +0,0 @@
-def setbit(byte, offset, value):
```
Comment (Kriechi) [inline mitmproxy/utils/bits.py]: No other parts in the entire code base use these functions.
And we didn't have test for it either...

File: mitmproxy/contentviews/base.py (hunk 1 above)
Comment (Kriechi) [inline mitmproxy/contentviews/base.py]: Yes this is kinda unclear. My intention was that there should be no code in `__init__.py` files, except for imports and the `__all__` thing. I wasn't aware we had a viewmanage - I will take a look and maybe move it there...

Comment (mhils) [inline mitmproxy/contentviews/base.py]: We don't have `viewmanager` yet,  but it'd make sense to create one for putting the stuff in. I generally agree on the move out of __init__ fwiw. :)

File: mitmproxy/contentviews/__init__.py [hunk 3]
```diff
@@ -173,11 +43,23 @@ def get_content_view(viewmode: View, data: bytes, **metadata):
 add(image.ViewImage())
 add(query.ViewQuery())
 
-if protobuf.ViewProtobuf.is_available():
+if protobuf.ViewProtobuf.is_available():  # pragma: no cover
```
Comment (Kriechi) [inline mitmproxy/contentviews/__init__.py]: Regardless of the protobuf-view quality, it was never available - I guess travis misses a package or something? Or does this only work if one manually installs the package?

**Comment by Kriechi** [comment]
Ok, I will refactor the individual-coverage check to be similarly configurable as the full-coverage task.

File: mitmproxy/contentviews/__init__.py (hunk 3 above)
Comment (mhils) [inline mitmproxy/contentviews/__init__.py]: The view shells out to `protoc`, which needs to be on PATH (this is what this function checks).
Default install does not add it to the PATH. We should use the protobuf library directly, so there's no excuse for hiding missing coverage here.

File: mitmproxy/utils/bits.py (hunk 2 above)
Comment (mhils) [inline mitmproxy/utils/bits.py]: I'd still say those are generic enough to be in their own module. 🙂

Comment (Kriechi) [inline mitmproxy/utils/bits.py]: Feel free to write tests, move them back again, and make it 100% covered. 😃

**Comment by Kriechi** [comment]
Refactor complete. Please take another look at this PR.
If needed, I could try and split out a few things to make the diff more readable...

**Comment by mhils** [comment]
Comments from https://github.com/mitmproxy/mitmproxy/pull/2011#pullrequestreview-21480384 are still unaddressed I think. :)

**Comment by Kriechi** [comment]
Coverage is already enforced - in a previous PR.
I will leave the extraction of a `viewmanager` for a future PR.
Anything still not addressed?

**Comment by mhils** [comment]
Status? :)

**Pull request closed without merging**